packages/shared/src/components/modals/SquadPostModal.tsx

```diff
@@ -0,0 +1,226 @@
+    inlineActions: true,
+  };
+
+  const postLinkProps = {
+    href: post.sharedPost.permalink,
+    title: 'Go to article',
+    target: '_blank',
+    rel: 'noopener',
+    onClick: onReadArticle,
+    onMouseUp: (event: React.MouseEvent) =>
+      event.button === 1 && onReadArticle(),
+  };
+
+  const getHeight = () => {
+    if (height === null) return 'auto';
+
+    return shoudShowSumamry ? height : 0;
+  };
+
+  return (
+    <BasePostModal
+      {...props}
+      size={Modal.Size.Large}
+      onRequestClose={onRequestClose}
+    >
+      {position === 'fixed' && (
+        <FixedPostNavigation
+          {...navigationProps}
+          className={{
+            container: modalSizeToClassName[ModalSize.Large],
+          }}
+        />
+      )}
+      <PostContentContainer
+        className="relative py-4 px-6 post-content"
+        hasNavigation={hasNavigation}
+      >
+        <PostNavigation
+          {...navigationProps}
+          onReadArticle={null}
+          className={{ actions: 'ml-auto', container: 'mb-6' }}
+        />
+        <SourceInfo
+          date={postDateFormat(post.createdAt)}
+          source={post.source}
+        />
+        <UserShortInfo
+          className="items-center pt-3"
+          imageSize="xxlarge"
+          user={post.author}
+        />
+        <p className="mt-6 typo-title3">
+          They breezed through simple tasks. I threw more complex tasks at them
+          — tasks for those who had a year of experience. What did I get?
+        </p>
+        <div className="flex flex-col mt-8 rounded-16 border border-theme-divider-tertiary">
+          <span className="flex flex-row p-4 max-w-full">
+            <div className="flex flex-col flex-1">
+              <h2 className="flex flex-wrap mb-4 font-bold line-clamp-2 typo-body">
+                {post.sharedPost.title}
+              </h2>
+              <SourceInfo
+                date={`${post.sharedPost.readTime}m read time`}
+                source={post.source}
+                typo="typo-footnote"
+                size="small"
+              />
+              <ReadArticleButton
+                buttonSize="medium"
+                className="mt-5 btn-secondary w-fit"
+                href={post.sharedPost.permalink}
+              />
+            </div>
+            <a
+              {...postLinkProps}
+              className="block overflow-hidden w-70 rounded-2xl cursor-pointer"
+            >
+              <LazyImage
+                imgSrc={post.image}
+                imgAlt="Post cover image"
+                ratio="52%"
+                eager
+                fallbackSrc="https://res.cloudinary.com/daily-now/image/upload/f_auto/v1/placeholders/1"
+              />
+            </a>
+          </span>
+          {/* {post.summary && showSumamry && ( */}
+          <PostSummary
+            ref={(el) => {
+              if (!el?.offsetHeight || height !== null) return;
+
+              setHeight(el.offsetHeight);
```

Comment:
Maybe make this a const above?

Reply:
We can't. Because we will be overriding the value based on the initial height. We set it to `0` when it is hidden.